packages/manager/src/features/linodes/LinodesDetail/LinodeNetworking/IPSharing.tsx

```diff
@@ -154,36 +186,97 @@ const IPSharingPanel: React.FC<CombinedProps> = (props) => {
   };
 
   const remainingChoices = (selectedIP: string): string[] => {
-    return ipChoices.filter((ip: string) => {
+    return Object.keys(ipChoices).filter((ip: string) => {
       const hasBeenSelected = ipsToShare.includes(ip);
       return ip === selectedIP || !hasBeenSelected;
     });
   };
 
   const onSubmit = () => {
-    const finalIPs = uniq(ipsToShare.filter(Boolean));
-
+    const groupedUnsharedRanges = {};
+    const finalIPs: string[] = uniq(
+      ipsToShare.reduce((previousValue, currentValue) => {
+        if (currentValue === undefined || currentValue === null) {
+          return previousValue;
+        }
+        const strippedIP: string = currentValue.split('/')[0];
+
+        // Filter out v4s and shared v6 ranges as only v6s and unshared ips will be added
+        const isStaticv6 = ipToLinodeID?.[currentValue]?.length === 1;
+        // For any IP in finalIPs that isn't shared (length of linode_ids === 1)
+        // make note in groupedUnsharedRanges so that we can first share that IP to
+        // the Linode it is statically routed to, then to the current Linode
+        if (isStaticv6) {
+          const linode_id = ipToLinodeID[currentValue][0];
+          if (groupedUnsharedRanges.hasOwnProperty(linode_id)) {
+            groupedUnsharedRanges[linode_id] = [
+              groupedUnsharedRanges[linode_id],
+            ].concat([strippedIP]);
```

Comment:
can we do this instead of the array concatenation?

```suggestion
            groupedUnsharedRanges[linode_id] = [...groupedUnsharedRanges[linode_id], [strippedIP]];
```

Reply:
good suggestion, this was actually a bug if there were more than 1 unshared range being shared